docs/layer_util.js

```diff
@@ -610,24 +621,47 @@ function removeScoreLayer() {
   redrawLayers();
 }
 
-const mapLoadingFinished = () => loadingElementFinished(mapContainerId);
+/** Notes that a layer has finished rendering on the map. */
+function mapLoadingFinished() {
+  loadingElementFinished(mapContainerId);
+}
 
 /**
  * Notes that an element has started loading, and add a handler to the Promise
- * to note when it finishes.
+ * to note when it finishes. Also handle errors if the Promise fails.
  * @param {Promise} promise
+ * @param {Object} layerInfoForErrors Data for layer in same format as coming
+ *     from Firestore. Must have `display-name` and `index` properties. Only
+ *     used in case of errors.
  * @return {Promise} wrappedPromise
  */
-function wrapPromiseLoadingAware(promise) {
+function wrapPromiseLoadingAware(promise, layerInfoForErrors) {
   addLoadingElement(mapContainerId);
-  return promise.then(mapLoadingFinished);
+  const {index, 'display-name': displayName} = layerInfoForErrors;
+  return promise
+      .catch((err) => {
+        showError(
```

Comment:
I wrote this in the other PR, what do you think about adding a little help icon so it's more clear you can interact (will maybe try to click but hover will become clear) with a messed up layer

Reply:
Don't think it's that useful here, because there's nothing you can do but reload the page. Especially now that your "globally accessible" PR is in. I guess we could add a section in the help to say to go to manage_layers.html and check that all your assets still exist? But until we have some idea what the user could do here, rather not add.